scripts/west_commands/zspdx/writer.py

```diff
@@ -42,6 +51,18 @@ def writeFileSPDX(f, bf):
             writeRelationshipSPDX(f, rln)
         f.write("\n")
 
+def generateDowloadUrl(url, revision):
+    # Only git is supported
+    # walker.py only parse revision if it's from git repositiory
+    if len(revision) == 0 or url.startswith("git://"):
+        return url
+
+    url = url.replace("https://", "git+https://")
+    url = url.replace("http://", "git+http://")
+    url = "@".join([url, revision])
```

Comment:
not sure this handle properly the case where url already starts with "git+http"? Shouldn't it?

Reply:
URL si extracted from meta file. The URL is the one for cloning the repository. I don't think it can have the url already starting with "git+".
This  function is not generic and takes into account that this script is run in zephyr: only git is supported, and url is parsed from ``git remote`` command (and maybe using ```west list``` values later)

So, do you think I should add a special case for the URL that starts with "git+" (even though  I don't think it can happen in our case) ?
Maybe you see a special case I did not see.